src/main/java/org/commcare/suite/model/AssertionSet.java

```diff
@@ -65,6 +65,24 @@ public Text getAssertionFailure(EvaluationContext ec) {
         }
     }
 
+    public Vector<String> getAssertionsXPaths() {
+        return this.xpathExpressions;
+    }
+
+    public Text evalAssertionAtIndex(Integer i, XPathExpression expression, EvaluationContext ec) {
+        try {
+            Object val = expression.eval(ec);
+            if (!FunctionUtils.toBoolean(val)) {
+                return messages.elementAt(i);
+            }
+        } catch (Exception e) {
+            // Following the patter from getAssertionFailure. Is there a reason not to throw the exception here?
```

Comment:
we are treating any exceptions evaulating an assertion as the assertion failure itself here and want to show the assertion message in case of these failures.

Reply:
I see, I was thinking it might be ambiguous if there was an error other than the assertion evaluating to `False`, but I guess you still know that it is an issue with that assertion.